internal/terraform/module/module_ops.go

```diff
@@ -440,20 +440,58 @@ func ParseVariables(ctx context.Context, fs ReadOnlyFS, modStore *state.ModuleSt
 
 	// TODO: Avoid parsing if the content matches existing AST
 
-	// TODO: Only parse the file that's being changed/opened, unless this is 1st-time parsing
-
 	// Avoid parsing if it is already in progress or already known
 	if mod.VarsDiagnosticsState[ast.HCLParsingSource] != op.OpStateUnknown && !job.IgnoreState(ctx) {
 		return job.StateNotChangedErr{Dir: document.DirHandleFromPath(modPath)}
 	}
 
-	err = modStore.SetVarsDiagnosticsState(modPath, ast.HCLParsingSource, op.OpStateLoading)
+	var files ast.VarsFiles
+	var diags ast.VarsDiags
+	rpcContext := lsctx.RPCContext(ctx)
```

Comment:
Depending on the order in which we merge https://github.com/hashicorp/terraform-ls/pull/1427 this may need to be updated. I'm happy to rebase my PR though.